x-pack/legacy/plugins/task_manager/lib/correct_deprecated_fields.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TaskInstance, TaskInstanceWithDeprecatedFields } from '../task';
+import { Logger } from '../types';
+
+export function ensureDeprecatedFieldsAreCorrected(
+  { id, taskType, interval, schedule, ...taskInstance }: TaskInstanceWithDeprecatedFields,
+  logger: Logger
+): TaskInstance {
+  if (interval) {
+    logger.warn(
+      `Task${
```

Comment:
are there tasks without id's?

Reply:
yup, when you schedule a new task it generally speaking doesn't have an ID, we generate a uuid in the SavedObjectsRepository.

Reply:
ah, makes sense ...